examples/jwt-signer-vault/main.rs

```diff
@@ -0,0 +1,112 @@
+use url::Url;
+use vaultrs::api::transit::requests::SignDataRequestBuilder;
+use vaultrs::api::transit::MarshalingAlgorithm;
+use vaultrs::client::{VaultClient, VaultClientSettingsBuilder};
+use vaultrs::transit::data::sign;
+
+use nr_auth::jwt::claims::Claims;
+
+/// A signed JWT should live enough for the System Identity Service to consume it.
+const DEFAULT_JWT_CLAIM_EXP: TimeDelta = TimeDelta::seconds(180);
+/// The "aud" (audience) claim identifies the recipients that the JWT is intended for.
+const DEFAULT_AUDIENCE: &str = "https://www.newrelic.com/";
+
+/// Main function to sign a JWT token with Vault
+/// It requires the following environment variables to be set:
+///
+/// VAULT_ADDRESS: Vault service address
+/// VAULT_TOKEN: Token to access Vault
+/// CLIENT_ID: Identity client_id
+/// TRANSIT_KEY_NAME: Transit Private key name
+///
+/// # Errors
+/// This function returns an error if:
+/// - The `.env` file is missing or cannot be loaded.
+/// - Required environment variables are not set.
+/// - The JWT signing process fails.
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Set the current directory to the example's path
+    let example_dir = Path::new("examples/jwt-signer-vault");
+    env::set_current_dir(&example_dir).expect("Failed to change directory");
+
+    tracing_subscriber::fmt()
+        .with_max_level(tracing::Level::INFO)
+        .init();
+
+    dotenv()
+        .map_err(|_| ".env file not found. Copy .env.dist file to .env and fill the variables")?;
+
+    let vault_addr = utils::os::env("VAULT_ADDRESS")?;
+    let token = utils::os::env("VAULT_TOKEN")?;
+    let key_name = utils::os::env("TRANSIT_KEY_NAME")?;
+    let client_id = utils::os::env("SYSTEM_IDENTITY_CLIENT_ID")?;
+
+    // Create a vault client
+    let client = VaultClient::new(
+        VaultClientSettingsBuilder::default()
+            .address(vault_addr)
+            .token(token)
+            .build()
+            .map_err(|e| format!("cannot build vault client settings {}", e))?,
+    )
+    .map_err(|e| format!("cannot build vault client {}", e))?;
+
+    // Build claims
+    let url = Url::from_str(DEFAULT_AUDIENCE).expect("constant valid url value");
+    let expires_at = Utc::now() + DEFAULT_JWT_CLAIM_EXP;
+    let timestamp = expires_at
+        .timestamp()
+        .try_into()
+        .map_err(|_| "converting token expiration time")?;
+    let claims = Claims::new(client_id, url, timestamp);
+
+    // Based on https://github.com/alexadamm/jwt-vault-go/blob/main/pkg/vault/vault.go#L245
+    let encoded_header = base64_encode(serde_json::to_vec(&Header::new(Algorithm::RS512))?);
+    let encoded_claims = base64_encode(serde_json::to_vec(&claims)?);
+    let input = format!("{}.{}", encoded_header, encoded_claims);
+    // Hash the input data
+    let mut hasher = Sha512::new();
+    hasher.update(input);
+    let digest = hasher.finalize().as_slice().to_vec();
+    // Encode the hash in Base64
```

Comment:
Nit: this comment feels redundant

Reply:
Fair enough. Removed in ed017cf3f5c6619a7cde9a17a116f136c6b43664